apps/docs/src/routes/playground/docgen.svelte.ts

```diff
@@ -0,0 +1,87 @@
+import { SvelteMap } from "svelte/reactivity";
+import { parse, createCacheStorage } from "svelte-docgen";
+import { ssp } from "sveltekit-search-params";
+import ts from "typescript";
+import * as tsvfs from "@typescript/vfs";
+
+// TODO: This can be removed once we migrate away from using `svelte2tsx`
+import shim from "svelte2tsx/svelte-shims-v4.d.ts?raw";
+
+export class Docgen {
+	#cache: ReturnType<typeof createCacheStorage>;
+
+	static #compiler_options = {
+		moduleDetection: ts.ModuleDetectionKind.Force,
+		target: ts.ScriptTarget.ES2023, // FIXME: createDefaultMapFromCDN does not support ESNext at the moment
+		module: ts.ModuleKind.ESNext,
+		esModuleInterop: true,
+		moduleResolution: ts.ModuleResolutionKind.Bundler,
+	} satisfies ts.CompilerOptions;
+
+	#fsmap = $state<SvelteMap<string, string>>(new SvelteMap());
```

Comment:
q: Does #fsmap really need to be a 'state'? At the moment, it doesn’t seem to serve any purpose.

Reply:
No, you're right. I don't remember why exactly I did put in `$state()` in the first place. I think I had an idea to provide a more detailed output in the _loading_ phase.

But... that would be over-engineering and definitely **not** a priority right now. Thanks for spotting it!